extensions/applicationinsights-dependencies-js/src/ajax.ts

```diff
@@ -3,25 +3,116 @@
+    IInstrumentHooksCallbacks, IInstrumentHook
 } from '@microsoft/applicationinsights-core-js';
-import { ajaxRecord } from './ajaxRecord';
+import { ajaxRecord, IAjaxRecordResponse } from './ajaxRecord';
 import { EventHelper } from './ajaxUtils';
 import { Traceparent } from './TraceParent';
+import dynamicProto from "@microsoft/dynamicproto-js";
+
+const AJAX_MONITOR_PREFIX = "ai.ajxmn.";
+const strDiagLog = "diagLog";
+const strThrowInternal = "throwInternal";
 
 let _isNullOrUndefined = CoreUtils.isNullOrUndefined;
+let _arrForEach = CoreUtils.arrForEach;
+
+// Using a global value so that to handle same iKey with multiple app insights instances (mostly for testing)
+let _markCount: number = 0;
+
+/** @Ignore */
+function _supportsFetch(): boolean {
+    let result: boolean = true;
+    let _global = getGlobal();
+    if (!_global || _isNullOrUndefined((_global as any).Request) ||
+        _isNullOrUndefined((_global as any).Request.prototype) ||
+        _isNullOrUndefined(_global.fetch)) {
+        result = false;
+    }
+    
+    return result;
+}
+
+/** 
+ * Determines whether ajax monitoring can be enabled on this document
+ * @returns True if Ajax monitoring is supported on this page, otherwise false
+ * @ignore
+ */
+function _supportsAjaxMonitoring(): boolean {
+    let result = false;
+
+    if (typeof XMLHttpRequest !== strUndefined && !_isNullOrUndefined(XMLHttpRequest)) {
+        let proto = XMLHttpRequest[strPrototype];
+        result = !_isNullOrUndefined(proto) &&
+            !_isNullOrUndefined(proto.open) &&
+            !_isNullOrUndefined(proto.send) &&
+            !_isNullOrUndefined(proto.abort);
+    }
+
+    // disable in IE8 or older (https://www.w3schools.com/jsref/jsref_trim_string.asp)
+    try {
+        " a ".trim();
+    } catch (ex) {
+        result = false;
+    }
+
+    return result;
+}
+
+/** @Ignore */
+function _getFailedAjaxDiagnosticsMessage(xhr: XMLHttpRequestInstrumented): string {
+    let result = "";
+    try {
+        if (!_isNullOrUndefined(xhr) &&
+            !_isNullOrUndefined(xhr.ajaxData) &&
+            !_isNullOrUndefined(xhr.ajaxData.requestUrl)) {
+            result += "(url: '" + xhr.ajaxData.requestUrl + "')";
+        }
+    } catch (e) { }
+
+    return result;
+}
+
+/** @ignore */
+function _throwInternalCritical(ajaxMonitorInstance:AjaxMonitor, msgId: _InternalMessageId, message: string, properties?: Object, isUserAct?: boolean): void {
+    ajaxMonitorInstance[strDiagLog]()[strThrowInternal](LoggingSeverity.CRITICAL, msgId, message, properties, isUserAct);
+}
+
+/** @ignore */
+function _throwInternalWarning(ajaxMonitorInstance:AjaxMonitor, msgId: _InternalMessageId, message: string, properties?: Object, isUserAct?: boolean): void {
+    ajaxMonitorInstance[strDiagLog]()[strThrowInternal](LoggingSeverity.WARNING, msgId, message, properties, isUserAct);
+}
```

Comment:
Can you please explain what does this syntax mean here `instance[]()[]()`? where is `ajaxMonitorInstance["dialog"]` come from? Thanks :)

Reply:
This 
instance["diagLog"]()["throwInternal"](....)
is effectively the same as this.
instance.diagLog().throwInternal(...)

The primary difference is that the first situation allows for better minification (only) when the quoted strings are used more than once as the variables holding the quoted strings can be minified to 1 (or 2 characters) while the direct names cannot.

I should also added that for minification you MUST use variables to hold the strings as using quote strings actually adds more characters than it removes, and uglify will actually convert direct quoted strings back to the full names anyway.